app/controllers/api/storage.php

```diff
@@ -84,41 +84,44 @@
         if (!$upload->isValid($file['tmp_name'])) {
             throw new Exception('Invalid file', 403);
         }
-
+        $localDevice = Storage::getDevice('self');
         // Save to storage
-        $size = $device->getFileSize($file['tmp_name']);
+        $size = $localDevice->getFileSize($file['tmp_name']);
         $path = $device->getPath(\uniqid().'.'.\pathinfo($file['name'], PATHINFO_EXTENSION));
         
-        if (!$device->upload($file['tmp_name'], $path)) { // TODO deprecate 'upload' and replace with 'move'
-            throw new Exception('Failed moving file', 500);
-        }
-
-        $mimeType = $device->getFileMimeType($path); // Get mime-type before compression and encryption
-
-        if (App::getEnv('_APP_STORAGE_ANTIVIRUS') === 'enabled') { // Check if scans are enabled
-            $antiVirus = new Network(App::getEnv('_APP_STORAGE_ANTIVIRUS_HOST', 'clamav'),
-                (int) App::getEnv('_APP_STORAGE_ANTIVIRUS_PORT', 3310));
-
-            if (!$antiVirus->fileScan($path)) {
-                $device->delete($path);
-                throw new Exception('Invalid file', 403);
+        
+        $mimeType = $localDevice->getFileMimeType($file['tmp_name']); // Get mime-type before compression and encryption, getting mime-type from local file to avoid making external request
+        
+        if(App::getEnv('_APP_STORAGE_DEVICE', Storage::DEVICE_LOCAL) === Storage::DEVICE_LOCAL) {
+            if (!$device->upload($file['tmp_name'], $path)) { // TODO deprecate 'upload' and replace with 'move'
```

Comment:
Is the comment still relevant? This might be the best opportunity to implement this.